packages/ubuntu_desktop_installer/lib/utils/product_info_extractor.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:io';
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:flutter/foundation.dart';
+
+class ProductInfoExtractor {
+  static String _cachedProductInfo = '';
+
+  final FileSystem fileSystem;
+
+  ProductInfoExtractor({
+    this.fileSystem = const LocalFileSystem(),
+  });
+
+  String getProductInfo({
+    String isoPath = '/cdrom/.disk/info',
+    String localPath = '/etc/os-release',
+  }) {
+    if (_cachedProductInfo.isNotEmpty) {
+      return _cachedProductInfo;
+    }
+
+    if (fileSystem.file(isoPath).existsSync()) {
+      try {
+        final content = fileSystem
+            .file(isoPath)
+            .readAsLinesSync()
+            .firstWhere((line) => line.trim().isNotEmpty);
+
+        _cachedProductInfo = content.substring(0, content.indexOf('"') - 1);
```

Comment:
A comment to explain the expected format (and why we're breaking on the first occurrence of ") would be helpful.